libs/core-ui/src/lib/Highchart/bubbleChart/buildScatterTemplate.ts

```diff
@@ -44,3 +48,21 @@ export function buildScatterTemplate(
   customData.Index = index;
   return customData;
 }
+
+function getAxisValueMapping(
+  axisProperty: string,
+  axisValue: number,
+  axisMap?: { [key: number]: string }
+): string | number {
+  if (axisMap && axisProperty === JointDataset.ClassificationError) {
+    const nearestValue = getNearestValue(axisValue, Object.keys(axisMap));
+    return axisMap[nearestValue];
+  }
+  return axisValue;
+}
+
+function getNearestValue(axisValue: number, axisMapKeys: any[]): number {
```

Comment:
Why do we need to get the nearest value. Can't we use the `x`/`y` to get the center and map that value?

Reply:
Scatterplot data is independent of bubble plot data at this time. As we don't persist bubble plot data, it is not possible to get x and y of clusters at this moment in scatter plot. There is a item to persist bubble plot data when axis doesn't change. This can be done as a follow up task of that item.

Reply:
Thanks Gaurav, updated to use x and y from cluster.